ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/TableController.java

```diff
@@ -199,13 +199,21 @@ public void upgradeHiveTable(Context ctx) {
     String amsUri = AmsUtil.getAMSThriftAddress(serviceConfig, Constants.THRIFT_TABLE_SERVICE_NAME);
     catalogMeta.putToCatalogProperties(CatalogMetaProperties.AMS_URI, amsUri);
     TableMetaStore tableMetaStore = ArcticCatalogUtil.buildMetaStore(catalogMeta);
+    // check whether catalog support MIXED_HIVE format.
+    Set<TableFormat> tableFormats = ArcticCatalogUtil.tableFormats(catalogMeta);
+    Preconditions.checkState(
+        tableFormats.contains(TableFormat.MIXED_HIVE),
+        "Catalog %s does not support MIXED_HIVE format",
+        catalog);
+    // we should only keep MIXED_HIVE format，
+    // so `CatalogLoader.createCatalog` can get right CatalogImpl through calling catalogImpl.
+    Map<String, String> catalogProperties = catalogMeta.getCatalogProperties();
+    catalogProperties.put(CatalogMetaProperties.TABLE_FORMATS, TableFormat.MIXED_HIVE.name());
+
```

Comment:
This will take effect to catalogMeta, I think it's better to create a copy of `catalogProperties`